spec/user_spec.rb

```diff
@@ -0,0 +1,191 @@
+require 'spec_helper'
+
+RSpec.describe User do
+  let(:base64_data) do
+    File.read(
+      File.expand_path(File.join(File.dirname(__FILE__), 'fixtures', 'generic_avatar'))
+    ).strip
+  end
+
+  let!(:rails_url) { Rails.application.routes.url_helpers }
+  let(:user)       { User.create }
+
+  context 'when user uses an avatar' do
+    context 'when user does not have an avatar attached yet' do
+      it 'does not have an avatar attached' do
+        expect(user.avatar.attached?).not_to be
+      end
+
+      context 'when \'user.avatar.attach\' is called' do
```

Comment:
don't escape single quotes, honestly I think it looks really bad. Maybe you can use 'when "user.avatar.attach" is called'